### ISSUE TITLE
Reduce native binary dependencies

### ISSUE THREAD

**Issue by gep13** [GitTools/GitVersion#1511]
@GitTools/developers what are your thoughts on taking this for a spin:

https://github.com/libgit2/libgit2sharp/pull/1618

It would reduce the native dependencies that we currently have, but off the top of my head, I am not sure whether it would "work" for us.

Not in a position to take this for a spin just now, but I thought I would bring it up when it found out about it.

**Comment by asbjornu** [comment]
Interesting. If @libgit2 produce a package for it, we can add it and run the tests. As long as all the tests are green, I would give this a huge 👍.

**Comment by asbjornu** [comment]
Interesting. If @libgit2 produce a package for it, we can add it and run the tests. As long as all the tests are green, I would give this a huge 👍.

**Comment by pascalberger** [comment]
Definitely interesting once available. But probably needs some time until it is available and stable enough for using in GitVersion

**Comment by JakeGinnivan** [comment]
GitVersion is already super chatty to git and this would likely slow things down heaps? I think we probably need to move to the in memory model first.

**Comment by ethomson** [comment]
> GitVersion is already super chatty to git and this would likely slow things down heaps? I think we probably need to move to the in memory model first.

When you say "super chatty", can you give me more details?  I see that you do a fetch (unless configured not to) but that's just the one call.  I _suspect_ that your fetch will be slower than native code, but if you're only doing the one call, I don't think it will be devastating.  (Local repository access is, of course, unchanged.)

**Comment by ethomson** [comment]
So trying a bit of a different approach at solving this problem, I've released a prerelease LibGit2Sharp that should help a bit here.  I've updated libgit2 to have built-in support for proxies, so we no longer rely upon libcurl.  I think this should help the native dependency problems quite a bit.

This is available in [LibGit2Sharp 0.26.0-preview-0070](https://www.nuget.org/packages/LibGit2Sharp/0.26.0-preview-0070).
